corehq/apps/case_search/xpath_functions/comparison.py

```diff
@@ -119,22 +116,19 @@ def _create_system_datetime_query(domain, case_property_name, op, value, node):
     in Asia/Seoul timezone begins at 2023-06-04T20:00:00 UTC.
     This might be inconsistent in daylight savings situations.
     """
+    if op == NEQ:
+        return filters.NOT(_create_system_datetime_query(domain, case_property_name, EQ, value, node))
+
     timezone = get_timezone_for_domain(domain)
     utc_equivalent_datetime_value = adjust_input_date_by_timezone(value_to_date(node, value), timezone, op)
-    if op in [EQ, NEQ]:
-        day_start_datetime = utc_equivalent_datetime_value
-        day_end_datetime = (utc_equivalent_datetime_value + timedelta(days=1))
-        op_value_dict = {
-            RANGE_OP_MAPPING[">="]: day_start_datetime,
-            RANGE_OP_MAPPING["<"]: day_end_datetime,
-        }
-        query = _case_property_range_query(case_property_name, op_value_dict, node)
```

Comment:
We know these are datetime properties, so I'm switching here to the datetime-specific filter, rather than the one that duck types